packages/flutter/lib/src/painting/borders.dart

```diff
@@ -30,16 +30,91 @@ enum BorderStyle {
 /// - [inside] provides padding with full [BorderSide.width].
 /// - [center] provides padding with half [BorderSide.width].
 /// - [outside] provides zero padding, as stroke is drawn entirely outside.
-enum StrokeAlign {
+@immutable
+class StrokeAlign {
+  /// This constructor allows any value between 0.0 and 1.0.
+  const StrokeAlign(this.value)
+      : assert(value != null),
+        assert(value >= 0.0, 'StrokeAlign must be greater than or equal to 0.0.'),
+        assert(value <= 1.0, 'StrokeAlign must be less than or equal to 1.0.');
+
   /// The border is drawn on the inside of the border path.
-  inside,
+  static const StrokeAlign inside = StrokeAlign(0.0);
```

Comment:
For me, it would be more intuitive to have negative one be "inside" and one be "outside". Then it's easy to know where it is relative to "center". But this works too, I just have to think of it as the proportion of the line that lies outside of the shape.

Reply:
Both are fine. I did this so it worked with lerp.

Can you help here? How would you do with -1, 0, 1?
<img width="761" alt="image" src="https://user-images.githubusercontent.com/351125/181121455-a42a0b75-2ec3-4e71-8d9e-a23502e08329.png">

Reply:
Here

Reply:
If the value goes from -1 to 1, then it's this:

```dart
// Inside == -1 => side.width
// Center == 0 => side.width * 0.5
// Outside == 1 => 0
EdgeInsets.all(side.width * (1 - (1 + side.strokeAlign.value) / 2));
```

Reply:
How do you solve this?

```dart
          final RRect borderRect = borderRadius.resolve(textDirection).toRRect(rect);
          final RRect inner = borderRect.deflate(ui.lerpDouble(side.width, 0, side.strokeAlign.value)!);
          final RRect outer = borderRect.inflate(ui.lerpDouble(0, side.width, side.strokeAlign.value)!);
          canvas.drawDRRect(outer, inner, paint);
```
From RoundedRectangleBorder. Needs to `deflate(side.width)` (inside), `deflate(side.width/2) AND inflate(side.width/2)` (center), `inflate(side.width)` outside.

Reply:
```dart
final RRect inner = borderRect.deflate(side.width * (1 - side.strokeAlign.value) / 2);
final RRect outer = borderRect.inflate(side.width * (side.strokeAlign.value + 1) / 2);
```

Reply:
The new formula is weird. It shrinks and grows

https://user-images.githubusercontent.com/351125/181131300-c43bcdf7-d38a-4402-b5b6-7ea8e0725cae.mov

Reply:
Ok, I got it working. I feel legibility went to the drain, but I'm trusting it works:
<img width="573" alt="image" src="https://user-images.githubusercontent.com/351125/181132237-3f0f4e43-b501-4707-8a9f-20ea6c2d1913.png">

Reply:
is there no lerp function that does that?

Reply:
I'm making my own.

Reply:
95% of the cases are the same lerps:
- lerp from zero to side.width
- lerp from side.width to zero

Now I understand what it is doing. Thanks.